backend/utilities/helpers/EnvHelper.py

```diff
@@ -1,16 +1,28 @@
 import os
 import logging
 from dotenv import load_dotenv
+from azure.identity import DefaultAzureCredential, get_bearer_token_provider
+from azure.keyvault.secrets import SecretClient
 
 logger = logging.getLogger(__name__)
 
 class EnvHelper:
     def __init__(self, **kwargs) -> None:
         load_dotenv()
+
+        self.AUTH_TYPE =os.environ.get("AUTH_TYPE")
+        if not self.AUTH_TYPE == 'rbac' and os.environ.get("USE_KEY_VAULT"):
+            self.credential = DefaultAzureCredential()
+            self.secret_client = SecretClient(os.environ.get("AZURE_KEY_VAULT_ENDPOINT"), self.credential)
+            self.AZURE_SEARCH_KEY = self.secret_client.get_secret(os.environ.get("AZURE_SEARCH_KEY")).value
+            self.AZURE_OPENAI_KEY = self.secret_client.get_secret(os.environ.get("AZURE_OPENAI_KEY")).value
+        else:
+            self.AZURE_SEARCH_KEY = None if self.AUTH_TYPE == 'rbac' else os.environ.get("AZURE_SEARCH_KEY")
+            self.AZURE_OPENAI_KEY = "" if self.AUTH_TYPE == 'rbac' else os.environ.get("AZURE_OPENAI_KEY")
+            
```

Comment:
I think this code is similar to code I've seen in this pr, can you see if there's a chance we can reuse this?

Reply:
This code is used in two places, but they are `front-end` and `back-end` modules, which are independent of each other, so they cannot be reused.